lib/cartopy/mpl/contour.py

```diff
@@ -20,66 +21,91 @@ class GeoContourSet(QuadContourSet):
+                # coordinates through to axes projection coordinates.
+                data_t = self.axes.transData
+                col_to_data = col.get_transform() - data_t
+
+                # Now that we have the transform, project all of this
+                # collection's paths.
+                new_paths = [col_to_data.transform_path(path)
+                             for path in paths]
+                new_paths = [path for path in new_paths
+                             if path.vertices.size >= 1]
+
+                # The collection will now be referenced in axes projection
+                # coordinates.
+                col.set_transform(data_t)
+
+                # Clear the now incorrectly referenced paths.
+                del paths[:]
+
+                for path in new_paths:
+                    if path.vertices.size == 0:
+                        # Don't persist empty paths. Let's get rid of them.
+                        continue
+
+                    # Split the path if it has multiple MOVETO statements.
+                    codes = np.array(
+                        path.codes if path.codes is not None else [0])
+                    moveto = codes == mpath.Path.MOVETO
+                    if moveto.sum() <= 1:
+                        # This is only one path, so add it to the collection.
+                        paths.append(path)
+                    else:
+                        # The first MOVETO doesn't need cutting-out.
+                        moveto[0] = False
+                        split_locs = np.flatnonzero(moveto)
+
+                        split_verts = np.split(path.vertices, split_locs)
+                        split_codes = np.split(path.codes, split_locs)
+
+                        for verts, codes in zip(split_verts, split_codes):
+                            # Add this path to the collection's list of paths.
+                            paths.append(mpath.Path(verts, codes))
+
+        else:
+            # Where contour paths exist at the edge of the globe, sometimes a
+            # complete path in data space will become multiple paths when
+            # transformed into axes or screen space.  Matplotlib's contour
+            # labelling does not account for this so we need to give it the
+            # pre-transformed paths to work with.
 
             # Define the transform that will take us from collection
             # coordinates through to axes projection coordinates.
             data_t = self.axes.transData
-            col_to_data = col.get_transform() - data_t
+            col_to_data = self.get_transform() - data_t
 
             # Now that we have the transform, project all of this
             # collection's paths.
+            paths = self.get_paths()
             new_paths = [col_to_data.transform_path(path) for path in paths]
-            new_paths = [path for path in new_paths if path.vertices.size >= 1]
+            paths[:] = new_paths
```

Comment:
`set_paths` is not implemented for the `ContourSet` although it is for other collections.  I wonder if we should raise a feature request, although this use-case is pretty niche.  I guess we would only have a problem if `get_paths` started returning a copy.

Reply:
Yeah, I think this could make sense. It does seem a bit suspect to rely on changing the returned object here like this.

Reply:
Feature request now at https://github.com/matplotlib/matplotlib/issues/26340.